aws/resource_aws_iot_certificate.go

```diff
@@ -28,26 +28,58 @@ func resourceAwsIotCertificate() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"certificate_pem": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"public_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"private_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
 		},
 	}
 }
 
 func resourceAwsIotCertificateCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).iotconn
 
-	log.Printf("[DEBUG] Creating certificate from csr")
-	out, err := conn.CreateCertificateFromCsr(&iot.CreateCertificateFromCsrInput{
-		CertificateSigningRequest: aws.String(d.Get("csr").(string)),
-		SetAsActive:               aws.Bool(d.Get("active").(bool)),
-	})
+	if d.Get("csr") != "" {
+		log.Printf("[DEBUG] Creating certificate from csr")
+		out, err := conn.CreateCertificateFromCsr(&iot.CreateCertificateFromCsrInput{
+			CertificateSigningRequest: aws.String(d.Get("csr").(string)),
+			SetAsActive:               aws.Bool(d.Get("active").(bool)),
+		})
 
-	if err != nil {
-		log.Printf("[ERROR] %s", err)
-		return err
-	}
-	log.Printf("[DEBUG] Created certificate from csr")
+		if err != nil {
+			log.Printf("[ERROR] %s", err)
+			return err
```

Comment:
Terraform will automatically log returned errors so the line above this is extraneous. We should also return some error context here for operators and code maintainers about when this error occurred, e.g.

```suggestion
			return fmt.Errorf("error creating IoT Certificate from CSR: %s", err)
```